service/ec2/customizations.go

```diff
@@ -1,15 +1,90 @@
 package ec2
 
 import (
+	"math/rand"
+	"sync"
 	"time"
 
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/aws/awsutil"
+	"github.com/aws/aws-sdk-go/aws/client"
 	"github.com/aws/aws-sdk-go/aws/endpoints"
 	"github.com/aws/aws-sdk-go/aws/request"
 )
 
+// lockedSource is a thread-safe implementation of rand.Source
+type lockedSource struct {
```

Comment:
This is used by dynamodb right? Lets just move this type into the SDK's /internal/ folder under some sync package maybe `/internal/sdkrand`